File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/calendar/DeliveryCalendarRepository.kt
package com.reguerta.user.domain.calendar

interface DeliveryCalendarDefaultDayReader {
    suspend fun getDefaultDeliveryDayOfWeek(): DeliveryWeekday?
}

interface DeliveryCalendarRepository : DeliveryCalendarDefaultDayReader {
    suspend fun getAllOverrides(): List<DeliveryCalendarOverride>
    suspend fun upsertOverride(override: DeliveryCalendarOverride): DeliveryCalendarOverride
    suspend fun deleteOverride(weekKey: String)
}

File: ios/Reguerta/Reguerta/Domain/Calendar/DeliveryCalendarRepository.swift
import Foundation

protocol DeliveryCalendarRepository: Sendable {
    func defaultDeliveryDayOfWeek(environment: SessionEnvironment) async throws -> DeliveryWeekday
    func allOverrides(environment: SessionEnvironment) async throws -> [DeliveryCalendarOverride]
    func upsertOverride(
        _ override: DeliveryCalendarOverride,
        environment: SessionEnvironment
    ) async throws -> DeliveryCalendarOverride
    func deleteOverride(weekKey: String, environment: SessionEnvironment) async throws
}
